vms/avm/vm_benchmark_test.go

```diff
@@ -70,3 +81,89 @@ func BenchmarkLoadUser(b *testing.B) {
 		})
 	}
 }
+
+// GetAllUTXOsBenchmark is a helper func to benchmark the GetAllUTXOs depending on the size
+func GetAllUTXOsBenchmark(b *testing.B, utxoCount int) {
+
+	vm := VM{}
+	vm.genesisCodec = codec.New(math.MaxUint32, 1<<20)
+	_ = vm.genesisCodec.RegisterType(&avax.TestAddressable{})
+	c := codec.New(math.MaxUint32, 1<<20)
+	_ = c.RegisterType(&secp256k1fx.TransferOutput{})
+
+	vm.codec = &codecRegistry{
+		genesisCodec:  vm.genesisCodec,
+		codec:         c,
+		index:         0,
+		typeToFxIndex: vm.typeToFxIndex,
+	}
+	vm.state = &prefixedState{
+		state: &state{State: avax.State{
+			Cache:        &cache.LRU{Size: stateCacheSize},
+			DB:           prefixdb.New([]byte{1}, memdb.New()),
+			GenesisCodec: vm.genesisCodec,
+			Codec:        c,
+		}},
+
+		tx:       &cache.LRU{Size: idCacheSize},
+		utxo:     &cache.LRU{Size: idCacheSize},
+		txStatus: &cache.LRU{Size: idCacheSize},
+
+		uniqueTx: &cache.EvictableLRU{Size: txCacheSize},
+	}
+
+	// #nosec G404
+	for i := 0; i < utxoCount; i++ {
+		utxo := &avax.UTXO{
+			UTXOID: avax.UTXOID{
+				TxID:        ids.GenerateTestID(),
+				OutputIndex: rand.Uint32(),
+			},
+			Asset: avax.Asset{ID: ids.ID{'y', 'e', 'e', 't'}},
+			Out: &secp256k1fx.TransferOutput{
+				Amt: 100000,
+				OutputOwners: secp256k1fx.OutputOwners{
+					Locktime:  0,
+					Addrs:     []ids.ShortID{addrs[0]},
+					Threshold: 1,
+				},
+			},
+		}
+
+		if err := vm.state.FundUTXO(utxo); err != nil {
+			b.Fatal(err)
+		}
+	}
+
+	addrsSet := ids.ShortSet{}
+	addrsSet.Add(addrs[0])
+
+	var (
+		err               error
+		notPaginatedUTXOs []*avax.UTXO
+	)
+
+	b.ResetTimer()
+	// Fetch all UTXOs older version
+	notPaginatedUTXOs, _, _, err = vm.getAllUTXOs(addrsSet, ids.ShortEmpty, ids.Empty)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	if len(notPaginatedUTXOs) != utxoCount {
+		b.Fatalf("Wrong number of utxos. Expected (%d) returned (%d)", utxoCount, len(notPaginatedUTXOs))
+	}
+
+}
+
+func Benchmark100GetUTXosExistingVersion(b *testing.B) {
+	GetAllUTXOsBenchmark(b, 100)
+}
+
+func Benchmark10000GetUTXosExistingVersion(b *testing.B) {
+	GetAllUTXOsBenchmark(b, 10000)
+}
+
+func Benchmark100000GetUTXosExistingVersion(b *testing.B) {
+	GetAllUTXOsBenchmark(b, 100000)
+}
```

Comment:
You could use a test vector here to make this 1 function rather than 3.

Reply:
100%